components/engine_traits/src/engine.rs

```diff
@@ -172,17 +172,13 @@ pub trait TabletFactory<EK>: TabletAccessor<EK> {
 
     /// Open a tablet by id and suffix from cache---that means it should already
     /// be opened.
-    fn open_tablet_cache(&self, id: u64, suffix: u64) -> Option<EK> {
-        if let Ok(engine) = self.open_tablet_raw(&self.tablet_path(id, suffix), false) {
-            return Some(engine);
-        }
-        None
-    }
+    fn open_tablet_cache(&self, id: u64, suffix: u64) -> Option<EK>;
 
     /// Open a tablet by id and any suffix from cache
-    fn open_tablet_cache_any(&self, id: u64) -> Option<EK> {
-        self.open_tablet_cache(id, 0)
-    }
+    fn open_tablet_cache_any(&self, id: u64) -> Option<EK>;
+
+    /// Open a tablet by id and latest available suffix from cache
+    fn open_tablet_cache_latest(&self, id: u64) -> Option<EK>;
```

Comment:
Why need this?

Reply:
How can we get a latest cache from factory with out it ?

Reply:
But why would you need to "get a latest cache` in the first place?

Reply:
I think it's a normal operation for us to get the current tablet for a region from factory.

Reply:
No, it's not. Generally, tablet index is provided to get specific tablet. Otherwise, it should use `CachedTablet` instead.

Reply:
So, why open_tablet_cache_any is needed?

Reply:
It's used in tests where index is not available. It doesn't matter whether the tablet is up to date, it's OK as long as it becomes up to date in the end.